use ethers::core::types::{ParseChainError, H512};

**Version**
├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   ├── ethers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   ├── ethers-addressbook v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-contract v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   ├── ethers-contract-abigen v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-contract-derive v1.0.2 (proc-macro) (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   │   ├── ethers-contract-abigen v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-providers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-etherscan v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-solc v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-middleware v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   ├── ethers-contract v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-etherscan v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-providers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-signers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607)
    │   │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-providers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-signers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   └── ethers-solc v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-etherscan v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-middleware v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-providers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-solc v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-etherscan v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-solc v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-addressbook v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-contract v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-etherscan v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-providers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   ├── ethers-solc v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   │   │   │   ├── ethers-core v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    ├── ethers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)
    │   ├── ethers v1.0.2 (https://github.com/gakonst/ethers-rs#228f9607) (*)

Does not compile with following error : 

error[E0432]: unresolved import `ethers::core::types::ParseChainError`
  --> relay/src/relay.rs:12:27
   |
12 | use ethers::core::types::{ParseChainError, H512};
   |                           ^^^^^^^^^^^^^^^
   |                           |
   |                           no `ParseChainError` in `types`
   |                           help: a similar name exists in the module: `ParseI256Error`

For more information about this error, try `rustc --explain E0432`.

--- a/ethers-core/src/types/chain.rs
+++ b/ethers-core/src/types/chain.rs
@@ -150,7 +150,7 @@ macro_rules! impl_try_from_numeric {
 
                 fn try_from(value: $primitive) -> Result<Self, Self::Error> {
                     if value.bits() > 64 {
-                        // `ParseChainError` only has a `number` field which has the same type
+                        // `TryFromPrimitiveError` only has a `number` field which has the same type
                         // as the `#[repr(_)]` attribute on the enum.
                         return Err(ParseChainError { number: value.low_u64() })
                     }

--- a/ethers-core/src/types/chain.rs
+++ b/ethers-core/src/types/chain.rs
@@ -9,7 +9,9 @@ use strum::{AsRefStr, EnumString, EnumVariantNames};
 
 // compatibility re-export
 #[doc(hidden)]
-pub use num_enum::{TryFromPrimitive, TryFromPrimitiveError as ParseChainError};
+pub use num_enum::{TryFromPrimitive, TryFromPrimitiveError};
+#[doc(hidden)]
+pub type ParseChainError = TryFromPrimitiveError<Chain>;
 
 // When adding a new chain:
 //   1. add new variant to the Chain enum;
@@ -136,7 +138,7 @@ macro_rules! impl_try_from_numeric {
     ($($native:ty)+ ; $($primitive:ty)*) => {
         $(
             impl TryFrom<$native> for Chain {
-                type Error = ParseChainError<Self>;
+                type Error = ParseChainError;
 
                 fn try_from(value: $native) -> Result<Self, Self::Error> {
                     (value as u64).try_into()
@@ -146,7 +148,7 @@ macro_rules! impl_try_from_numeric {
 
         $(
             impl TryFrom<$primitive> for Chain {
-                type Error = ParseChainError<Self>;
+                type Error = ParseChainError;
 
                 fn try_from(value: $primitive) -> Result<Self, Self::Error> {
                     if value.bits() > 64 {
@@ -170,7 +172,7 @@ impl From<Chain> for u64 {
 impl_into_numeric!(u128 U64 U128 U256 U512);
 
 impl TryFrom<U64> for Chain {
-    type Error = ParseChainError<Self>;
+    type Error = ParseChainError;
 
     fn try_from(value: U64) -> Result<Self, Self::Error> {
         value.low_u64().try_into()

--- a/ethers-core/src/types/chain.rs
+++ b/ethers-core/src/types/chain.rs
@@ -1,5 +1,4 @@
 use super::{U128, U256, U512, U64};
-use num_enum::{TryFromPrimitive, TryFromPrimitiveError};
 use serde::{Deserialize, Serialize, Serializer};
 use std::{
     convert::{TryFrom, TryInto},
@@ -7,6 +6,10 @@ use std::{
     time::Duration,
 };
 use strum::{AsRefStr, EnumString, EnumVariantNames};
+
+// compatibility re-export
+#[doc(hidden)]
+pub use num_enum::{TryFromPrimitive, TryFromPrimitiveError as ParseChainError};
 
 // When adding a new chain:
 //   1. add new variant to the Chain enum;
@@ -133,7 +136,7 @@ macro_rules! impl_try_from_numeric {
     ($($native:ty)+ ; $($primitive:ty)*) => {
         $(
             impl TryFrom<$native> for Chain {
-                type Error = TryFromPrimitiveError<Self>;
+                type Error = ParseChainError<Self>;
 
                 fn try_from(value: $native) -> Result<Self, Self::Error> {
                     (value as u64).try_into()
@@ -143,13 +146,13 @@ macro_rules! impl_try_from_numeric {
 
         $(
             impl TryFrom<$primitive> for Chain {
-                type Error = TryFromPrimitiveError<Self>;
+                type Error = ParseChainError<Self>;
 
                 fn try_from(value: $primitive) -> Result<Self, Self::Error> {
                     if value.bits() > 64 {
-                        // `TryFromPrimitiveError` only has a `number` field which has the same type
+                        // `ParseChainError` only has a `number` field which has the same type
                         // as the `#[repr(_)]` attribute on the enum.
-                        return Err(TryFromPrimitiveError { number: value.low_u64() })
+                        return Err(ParseChainError { number: value.low_u64() })
                     }
                     value.low_u64().try_into()
                 }
@@ -167,7 +170,7 @@ impl From<Chain> for u64 {
 impl_into_numeric!(u128 U64 U128 U256 U512);
 
 impl TryFrom<U64> for Chain {
-    type Error = TryFromPrimitiveError<Self>;
+    type Error = ParseChainError<Self>;
 
     fn try_from(value: U64) -> Result<Self, Self::Error> {
         value.low_u64().try_into()

